consoles/sshVirtsh.pm

```diff
@@ -522,7 +522,22 @@ __END"
     # define the new domain
     $self->run_cmd("virsh $remote_vmm define $xmlfilename") && die "virsh define failed";
     if ($self->vmm_family eq 'vmware') {
-        $self->run_cmd('echo bios.bootDelay = \"10000\" >> /vmfs/volumes/datastore1/openQA/' . $self->name . '.vmx', domain => 'sshVMwareServer');
+        my $vmx = sprintf('/vmfs/volumes/datastore1/openQA/%s.vmx', $self->name);
+
+        # set default boot delay
+        $self->run_cmd("echo bios.bootDelay = \"10000\" >> $vmx", domain => 'sshVMwareServer');
+
+        # inject cloud init metadata and userdata required for the image if there are any
+        my $ci_meta = $bmwqemu::vars{CLOUD_INIT_META};
+        my $ci_user = $bmwqemu::vars{CLOUD_INIT_USER};
+        my $ci_encoding = $bmwqemu::vars{CLOUD_INIT_ENCODING};
```

Comment:
please either die if the according variables are not set or provide a good default.

Reply:
Sorry, why would you like to add `die`? VMs should be also tested w/o cloud-init as well

Reply:
But if you don't define those variables then undefined variables will be used in string concatenations causing at least Perl warnings if not further weird problems

Reply:
If I am not mistaken, `undef` is a falsy value, therefore the code should not get into the below condition and attempt to do string concat.

So it should behave like this, or no?

```
No warning in condition
```

```
#!/usr/bin/perl

use strict;
use warnings;

my $a = undef;

unless ($a) {
    print "No warning in condition\n";
}

if ($a) {
    print "Nothing\n";
}
```

Reply:
Yes, the if-block should do the trick.

Reply:
Yes, true. Sorry about that. Haven't looked two lines below :D